chainntnfs/mempool.go

```diff
@@ -19,7 +20,11 @@ type MempoolNotifier struct {
 
 	// subscribedInputs stores the inputs that we want to watch their
 	// spending event for.
-	subscribedInputs lnutils.SyncMap[wire.OutPoint, *MempoolSpendEvent]
+	subscribedInputs *lnutils.SyncMap[wire.OutPoint,
+		*lnutils.SyncMap[uint64, *MempoolSpendEvent]]
```

Comment:
I herd you like sync maps...

May want to consider using a `type` def (or just a type alias to reduce short hand) to make this a bit more readable. I guess we also need to update our style guide when it comes to the new generics syntax.

Reply:
tried to define a type before,
```go
type subscribersMap lnutils.SyncMap[uint64, *MempoolSpendEvent]
```

but when I load it from the `subscribedInputs`, it lost access to `lnutils.SyncMap`, very weird...for instance, the following call on `ForEach` will fail, and `subs` is a `sync.Map`, not `lnutils.SyncMap`.
```go
// Load the subscriber.
subs, loaded := m.subscribedInputs.Load(op)
if !loaded {
	Log.Errorf("Sub not found for %s", op)
	return
}

// Iterate all the subscribers for this input and notify them.
subs.ForEach(func(id uint64, sub *MempoolSpendEvent) error {
	m.wg.Add(1)
	go notifySingle(id, sub, op, detail)

	return nil
})
```